components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/TokenManagementDAOImpl.java

```diff
@@ -424,6 +424,44 @@ public void revokeOAuthConsentByApplicationAndUser(String username, String tenan
         }
     }
 
+    /**
+     * Revoke the OAuth Consent which is recorded in the IDN_OPENID_USER_RPS table against the Application
+     *
+     * @param applicationName - Name of the OAuth App
+     * @throws IdentityOAuth2Exception - If an unexpected error occurs.
+     */
+    @Override
+    public void revokeOAuthConsentByApplication(String applicationName) throws IdentityOAuth2Exception {
```

Comment:
application name is not unique across tenants?
If we have two apps with the same name in two tenants, this would remove the consents of both apps.

Shall we use the application's resource ID (uuid identifier) or application name + tenant domain

Reply:
Addressed in [40fde89](https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1627/commits/40fde89cb099493c8f2a302b34b1141d9aea3e8d)

Reply:
@mefarazath @dewniMW 
By observing, the approach of deleting the consent by the app_name and the app_tenant_domain do not seem to delete all the consent records related to a SaaS application, since the **TENANT_ID** in the **IDN_OPENID_USER_RPS** table stores the tenant id of the user.

To address that, In [5ad8177](https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1627/commits/5ad8177d12821d40918569e36e0a9eca4ff0f4d1) revoke method has been updated to delete the consent by passing an array of tenant domains of authorised users. But then again, there’s a chance of having a performance hit since we are trying to delete the consent in an iterative manner. (we can avoid that if there’s another field in IDN_OPENID_USER_RPS table to uniquely identify the application)